tests/python/relay/test_op_level3.py

```diff
@@ -45,6 +45,31 @@ def test_transpose_infer_type():
         (t, n, 100), "float32")
 
 
+def test_squeeze_default_axes_infer_type():
+    ib = relay.ir_builder.IRBuilder()
+    n, t, d = 1, 4, 1
+    x = ib.param("x", relay.ty.TensorType((n, t, d), "float32"))
+    with ib.function(x) as func:
+        ib.ret(relay.squeeze(x))
+    ib.ret(func)
+    func = relay.ir_pass.infer_type(ib.env, func.to_func())
+    ftype = func.checked_type
+    assert ftype.ret_type == relay.ty.TensorType(
+        (4,), "float32")
+
+def test_squeeze_axes_infer_type():
+    ib = relay.ir_builder.IRBuilder()
+    n, t, d = 1, 4, 1
+    x = ib.param("x", relay.ty.TensorType((n, t, d), "float32"))
+    with ib.function(x) as func:
+        ib.ret(relay.squeeze(x, axes=(2,)))
+    ib.ret(func)
+    func = relay.ir_pass.infer_type(ib.env, func.to_func())
+    ftype = func.checked_type
+    assert ftype.ret_type == relay.ty.TensorType(
+        (1, 4), "float32")
+
+
```

Comment:
I suggest adding a case where the chosen axis has the wrong dimension.

Reply:
how to test for failure? just catch?

Reply:
nose has an `@raises` decorator

e.g.
```python
from nose.tools import raises
...
@raises(YourException)
def test_foo():
    ...
```